### PR TITLE
Add Icon to Windows native binary

### LINKED ISSUE #1086: Add Icon to Windows native binary
The native exe generated on Windows platform doesn't have an icon on it. We need to add either the default or user provided icon to it.

Related to https://github.com/gluonhq/gluonfx-maven-plugin/issues/298

### PR DESCRIPTION
**Pull request by abhinayagarwal**
Taking inspiration from @freya022's  [comment](https://github.com/gluonhq/gluonfx-maven-plugin/issues/298#issuecomment-824352685), I have been able to add an icon to the native exe by linking an icon object.

To create the icon object, we have a static file.  Use `rc.exe` command to compile the resource file and then `cvtres.exe` to convert the output file into an linkable object. 

### Issue

Fixes #1086 

### Progress

- [x] Change must not contain extraneous whitespace
- [x] License header year is updated, if required
- [x] Verify the contributor has signed [Gluon Individual Contributor License Agreement (CLA)](https://docs.google.com/forms/d/16aoFTmzs8lZTfiyrEm8YgMqMYaGQl0J8wA0VJE2LCCY)

### CODE REVIEW

File: src/main/java/com/gluonhq/substrate/target/WindowsTargetConfiguration.java [hunk 1]
```diff
@@ -217,4 +217,68 @@ public boolean packageApp() throws IOException, InterruptedException {
...
+    }
+
+    // During development if user changes the application icon, the same is not reflected immediately in Explorer.
+    // To fix this, a cache clearance of the Windows explorer is required.
+    private void clearExplorerCache() throws IOException, InterruptedException {
+        ProcessRunner clearCache;
+        String osName = System.getProperty("os.name").toLowerCase(Locale.ROOT);
+        // For Windows version < 10, use `ie4uinit.exe -ClearIconCache`
+        if (osName.equals("windows 10") || osName.equals("windows 11")) {
+            clearCache = new ProcessRunner("ie4uinit", "-show");
```
Comment (jperedadnr) [inline src/main/java/com/gluonhq/substrate/target/WindowsTargetConfiguration.java:241]: You can do `ProcessRunner clearCache = new ProcessRunner("ie4uinit");`
And then, with an `if` or a ternary expression:
`clearCache.addArg(10+? "-show" : "-ClearIconCache");`

File: src/main/java/com/gluonhq/substrate/target/WindowsTargetConfiguration.java [hunk 2]
```diff
@@ -217,4 +217,66 @@ public boolean packageApp() throws IOException, InterruptedException {
...
+        if (Files.exists(userAssets) && Files.isDirectory(userAssets) && Files.exists(userAssets.resolve("icon.ico"))) {
+            FileOps.copyFile(userAssets.resolve("icon.ico"), tmpIconDir.resolve("icon.ico")) ;
+            Logger.logDebug("User provided icon.ico image used as application icon.");
+        } else {
+            Path windowsGenSrcPath = paths.getGenPath().resolve(sourceOS);
+            Path windowsAssetPath = windowsGenSrcPath.resolve(Constants.WIN_ASSETS_FOLDER);
+            Files.createDirectories(windowsAssetPath);
+            FileOps.copyResource("/native/windows/assets/icon.ico", windowsAssetPath.resolve("icon.ico"));
+            FileOps.copyFile(windowsAssetPath.resolve("icon.ico"), tmpIconDir.resolve("icon.ico"));
+            Logger.logInfo("Default icon.ico image used. " +
```
Comment (jperedadnr) [inline src/main/java/com/gluonhq/substrate/target/WindowsTargetConfiguration.java:265]: The standard message is:
```
Logger.logInfo("Default icon.ico image generated in " + windowsAssetPath + ".\n" +
                    "Consider copying it to " + rootPath + " before performing any modification");
```

**Review by jperedadnr — APPROVED** [review]

**Comment by westinyang** [comment]
How should it be used? Creating a gluonfx app, where should custom icons be placed?

**Comment by simeonlin** [comment]
If it helps someone: 
Put your File under `src\windows\assets\icon.ico`

(according westinyang's question)

**Pull request merged**